clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/images/preload/FilePreloaderExecutors.kt

```diff
@@ -57,8 +57,9 @@ internal class FilePreloaderExecutors @JvmOverloads constructor(
 
         for (url in urlMetas) {
             val task = executor.ioTaskNonUi<Unit>()
+            task.addOnSuccessListener { countDownLatch.countDown() }
+            task.addOnFailureListener { countDownLatch.countDown() }
```

Comment:
thread switching happens when we add success and failure listener, either fix that or you can move it to end of 
`task.execute("tag") {)` block

Reply:
fixed in https://github.com/CleverTap/clevertap-android-sdk/commit/d9ad2fb0dd5f0f84999732a0e2da95e72fce0866